app/src/main/java/com/example/homework6/mvvm/ListFromRetrofitViewModel.kt

```diff
@@ -0,0 +1,66 @@
+package com.example.homework6.mvvm
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.homework6.model.LceState
+import com.example.homework6.model.data.Person
+import com.example.homework6.model.repositories.PersonRepositoryImpl
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class ListFromRetrofitViewModel(private val personRepository: PersonRepositoryImpl) : ViewModel() {
+
+    private val _lceFlow = MutableStateFlow<LceState<List<Person>>>(LceState.Loading)
+    val lceFlow: Flow<LceState<List<Person>>> = _lceFlow.asStateFlow()
+
+    val _searchQueryFlow = MutableStateFlow("")
+    val searchQueryFlow = _searchQueryFlow.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            personRepository.getPersonListFromApi()
+                .fold(
+                    onSuccess = {
+                        _lceFlow.tryEmit(LceState.Content(persons = it))
+                    },
+                    onFailure = {
+
+                    }
+                )
```

Comment:
используешь только ветку success, проще было вот так тогда
```suggestion
                .onSuccess {
                     _lceFlow.tryEmit(LceState.Content(persons = it))
                }
```
а в целом я бы сделал все через одну цепочку
```
    searchFlow
        .debounce(100)
        .mapLatest {
             personRepository.getPersonListFromApi().fold(...)
        }
```